### PR TITLE
[AFF] Allow object handler to serialize JSON via experimental feature

### PR DESCRIPTION
**Pull request by albertmink**
Now, as we refactored the serialization in #5668, we want to serialize JSON via the experimental feature.

Filling the data with meaningful values will be the next step for us.

First steps towards #5658

### CODE REVIEW

**Comment by albertmink** [comment]
Result of the serialization (left), after being deserialized into the ABAP system. Reference AFF file on the right.
image

Example repo https://github.com/albertmink/testing_intf_aff

File: src/objects/zcl_abapgit_object_intf.clas.abap [hunk 1]
```diff
@@ -15,26 +15,26 @@ CLASS zcl_abapgit_object_intf DEFINITION PUBLIC FINAL INHERITING FROM zcl_abapgi
         zcx_abapgit_exception .
     METHODS deserialize_abap
       IMPORTING
-        !ii_xml     TYPE REF TO zif_abapgit_xml_input
-        !iv_package TYPE devclass
+        ii_xml     TYPE REF TO zif_abapgit_xml_input
+        iv_package TYPE devclass
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_intf.clas.abap:19]: no need to fix these unrelated lines, someone opening the code in SE80 will revert it

**Review by larshp — CHANGES_REQUESTED** [review]
on a general note, looking a this, we need to define what the internal representation of the object is, abstract it, and have two different ways for reading data into that representation

**Comment by albertmink** [comment]
will pause here and the refactoring first. First fill the ABAP data type, then put it to XML/JSON(experimental flag)

File: src/objects/zcl_abapgit_object_intf.clas.abap [hunk 2]
```diff
@@ -495,7 +577,141 @@ CLASS zcl_abapgit_object_intf IMPLEMENTATION.
...
+    SELECT SINGLE descript AS description category clsproxy AS proxy unicode AS abap_language_version
+           FROM vseointerf
+           INTO (rs_intf_aff-header-description, rs_intf_aff-category, rs_intf_aff-proxy,
+                 rs_intf_aff-header-abap_language_version)
+           WHERE clsname = is_interface_key-clsname AND version = '1' AND
+                 langu = rs_intf_aff-header-original_language.
+
+    rs_intf_aff-descriptions = cl_oo_aff_clif_helper=>get_descriptions_compo_subco(
+                              language  = rs_intf_aff-header-original_language
+                              clif_name = is_interface_key-clsname ).
```
[the remark this answers is not in the document]
Comment (albertmink) [reply src/objects/zcl_abapgit_object_intf.clas.abap]: indeed, its been removed in this PR.

File: src/objects/zcl_abapgit_object_intf.clas.abap [hunk 3]
```diff
@@ -324,14 +329,33 @@ CLASS zcl_abapgit_object_intf IMPLEMENTATION.
...
+      TRY.
+
+          lv_json_xstring = lo_ajson->serialize( ls_aff_intf ).
+
+          zif_abapgit_object~mo_files->add_raw( iv_ext = 'json'
+                                                iv_data = lv_json_xstring ).
+
+        CATCH cx_root INTO lx_exception.
+          zcx_abapgit_exception=>raise_with_text( lx_exception ).
+      ENDTRY.
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_intf.clas.abap:347]: put all this into the local class? a static method is okay from my POV

as things evolve, we might need to change this part, putting it to the local class reduces that impact

Comment (larshp) [inline src/objects/zcl_abapgit_object_intf.clas.abap:347]: put all this into the local class? a static method is okay from my POV

as things evolve, we might need to change this part, putting it to the local class reduces that impact

File: src/objects/zcl_abapgit_object_intf.clas.abap [hunk 4]
```diff
@@ -290,6 +290,8 @@ CLASS zcl_abapgit_object_intf IMPLEMENTATION.
     DATA:
       ls_intf             TYPE ty_intf,
       ls_clskey           TYPE seoclskey,
+      lx_exception        TYPE REF TO cx_root,
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_intf.clas.abap:293]:
```suggestion
```

**Review by larshp — APPROVED** [review]

**Pull request merged**